nifi-commons/nifi-record/src/main/java/org/apache/nifi/serialization/record/RecordFieldType.java

```diff
@@ -364,4 +364,8 @@ public boolean isWiderThan(final RecordFieldType fieldType) {
     public static RecordFieldType of(final String typeString) {
       return SIMPLE_NAME_MAP.get(typeString);
     }
+
+    public Set<RecordFieldType> getNarrowDataTypes() {
+        return Collections.unmodifiableSet(narrowDataTypes);
```

Comment:
Should make the member variable itself immutable instead of creating a new Set object to wrap the existing one for each invocation.

Reply:
Good point, I did fix this. I added only to the constructor which sets a non-empty Set as with empty set: it would be both confusing and both unnecessary: emptySet method return with implementation will not allow adding (remove is allowed but no item contained)